clisops/core/subset.py

```diff
@@ -737,23 +739,39 @@ def subset_shape(
 
     mask_2d = create_mask(
         x_dim=ds_copy.lon, y_dim=ds_copy.lat, poly=poly, wrap_lons=wrap_lons
-    )
+    ).clip(
```

Comment:
For cases where users use `create_mask` directly, would it make more sense to apply this fix to the code of `create_mask` ?

Reply:
Hum, I don't think so. This part is not really a fix, rather a quick-and-easy way so that the lines below work. i.e.:
```
        left = mask_2d.bfill(dim).sum(sp_dims - {dim})
        right = mask_2d.ffill(dim).sum(sp_dims - {dim})
        # True in the inner zone, False in the outer
        inner_mask = inner_mask & (left != 0) & (right != 0)
```
Without the `clip` the mask would have 0s on the zeroth shape and the `!= 0` test wouldn't make sense. I found that clip was faster (~ µs) than transforming the int `mask_2d` to a boolean mask (~ ms), that is all!

Reply:
Ah, OK. The fix is a bit esoteric to my limited depth of xarray knowledge but if it seems to be handling the issue, works for me.